### ISSUE TITLE
Invalid datetime format: 1292

### ISSUE THREAD

**Issue by numerfolt** [owncloud/music#1066]
### Steps to reproduce
1. Install Music-App in Nextcloud
2. Install playSub on iOS
3. Play any music so it gets the "last played" attribute

### Expected behaviour
No errors in logfile

### Actual behaviour
There are lots of errors in my logfile regarding "Invalid datetime format: 1292"

### Server configuration
**Operating system**: Linux nextcloudpi 6.1.21-v8+ aarch64 GNU/Linux

**Web server:** Apache/2.4.56 (Debian)

**Database:** 10.5.19-MariaDB-0+deb11u2 Debian 11

**PHP version:** PHP 8.1.18

**ownCloud version:** Nextcloud 26.0.1.1

**Updated from an older ownCloud or fresh install:** Update from older version but problem existing ever since first install

**Where did you install ownCloud from:** Nextcloudpi image

**Signing status (ownCloud 9.0 and above):** -

```
No errors found.
```

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'passwordsalt' => '***',
  'secret' => '***',
  'trusted_domains' => 
  array (
  ***
  ),
  'datadirectory' => '/media/dataDrive/ncdata',
  'dbtype' => 'mysql',
  'version' => '26.0.1.1',
  'overwrite.cli.url' => 'https://nextcloudpi/',
  'dbname' => 'nextcloud',
  'dbhost' => 'localhost',
  'dbport' => '',
  'dbtableprefix' => 'oc_',
  'mysql.utf8mb4' => true,
  'dbuser' => '***',
  'dbpassword' => '***',
  'installed' => true,
  'instanceid' => '***',
  'memcache.local' => '\\OC\\Memcache\\Redis',
  'memcache.locking' => '\\OC\\Memcache\\Redis',
  'redis' => 
  array (
    'host' => '/var/run/redis/redis.sock',
    'port' => 0,
    'timeout' => 0.0,
    'password' => '***',
  ),
  'tempdirectory' => '/media/dataDrive/ncdata/tmp',
  'mail_sendmailmode' => 'smtp',
  'mail_smtphost' => '***',
  'mail_smtpauth' => 1,
  'mail_smtpport' => '465',
  'mail_smtpname' => '***',
  'mail_smtppassword' => '***',
  'mail_smtpsecure' => 'ssl',
  'preview_max_x' => '2048',
  'preview_max_y' => '2048',
  'jpeg_quality' => '60',
  'overwriteprotocol' => 'https',
  'trusted_proxies' => 
  array (
    ***
  ),
  'maintenance' => false,
  'logfile' => '/media/dataDrive/ncdata/nextcloud.log',
  'loglevel' => '2',
  'log_type' => 'file',
  'htaccess.RewriteBase' => '/',
  'mail_from_address' => '***',
  'mail_smtpmode' => 'smtp',
  'mail_domain' => '***',
  'default_phone_region' => '***',
  'data-fingerprint' => '***',
  'memories.exiftool' => '/var/www/nextcloud/apps/memories/exiftool-bin/exiftool-aarch64-glibc',
  'memories.gis_type' => 1,
  'memories.vod.path' => '/var/www/nextcloud/apps/memories/exiftool-bin/go-vod-aarch64',
  'app_install_overwrite' => 
  array (
    0 => 'apporder',
  ),
  'theme' => '',
);
```

**List of activated apps:**

```
Enabled:
  - activity: 2.18.0
  - bookmarks: 13.0.1
  - calendar: 4.3.4
  - circles: 26.0.0
  - cloud_federation_api: 1.9.0
  - comments: 1.16.0
  - contacts: 5.2.0
  - contactsinteraction: 1.7.0
  - cookbook: 0.10.2
  - cospend: 1.5.10
  - dashboard: 7.6.0
  - dav: 1.25.0
  - federatedfilesharing: 1.16.0
  - files: 1.21.1
  - files_pdfviewer: 2.7.0
  - files_rightclick: 1.5.0
  - files_sharing: 1.18.0
  - files_trashbin: 1.16.0
  - files_versions: 1.19.1
  - firstrunwizard: 2.15.0
  - logreader: 2.11.0
  - lookup_server_connector: 1.14.0
  - maps: 1.0.2
  - music: 1.8.3
  - nextcloud_announcements: 1.15.0
  - nextcloudpi: 0.0.1
  - notes: 4.7.2
  - notifications: 2.14.0
  - notify_push: 0.6.3
  - oauth2: 1.14.0
  - password_policy: 1.16.0
  - photos: 2.2.0
  - previewgenerator: 99.99.99
  - privacy: 1.10.0
  - provisioning_api: 1.16.0
  - recommendations: 1.5.0
  - related_resources: 1.1.0-alpha1
  - serverinfo: 1.16.0
  - settings: 1.8.0
  - sharebymail: 1.16.0
  - side_menu: 3.7.4
  - survey_client: 1.14.0
  - systemtags: 1.16.0
  - tasks: 0.15.0
  - text: 3.7.2
  - theming: 2.1.1
  - twofactor_backupcodes: 1.15.0
  - updatenotification: 1.16.0
  - user_status: 1.6.0
  - viewer: 1.10.0
  - weather_status: 1.6.0
  - workflow_ocr: 1.26.1
  - workflowengine: 2.8.0
Disabled:
  - admin_audit: 1.16.0
  - bruteforcesettings: 2.6.0
  - encryption: 2.14.0
  - federation: 1.16.0 (installed 1.16.0)
  - files_external: 1.18.0
  - memories: 5.1.0 (installed 5.1.0)
  - news: 21.2.0 (installed 21.2.0)
  - support: 1.9.0 (installed 1.9.0)
  - suspicious_login: 4.4.0
  - tables: 0.5.0 (installed 0.5.0)
  - twofactor_totp: 8.0.0-alpha.0
  - user_ldap: 1.16.0
```

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

### Client configuration
**Browser:** Firefox

**Operating system:** Garuda Linux

#### Web server error log
```
[Mon May 22 01:02:33.542977 2023] [proxy_fcgi:error] [pid 179936:tid 547222425984] [client 87.152.111.211:58498] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Mon May 22 01:02:43.887716 2023] [proxy_fcgi:error] [pid 179937:tid 548078088576] [client 87.152.111.211:58502] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Mon May 22 01:02:46.951231 2023] [proxy_fcgi:error] [pid 179937:tid 547943870848] [client 87.152.111.211:58502] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Mon May 22 01:04:48.617594 2023] [proxy_fcgi:error] [pid 179937:tid 547927085440] [client 87.152.111.211:58506] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Mon May 22 01:04:48.803454 2023] [proxy_fcgi:error] [pid 179937:tid 548052910464] [client 87.152.111.211:58506] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Mon May 22 10:05:23.315079 2023] [proxy_fcgi:error] [pid 179937:tid 548069695872] [client 87.152.111.211:58627] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Mon May 22 10:45:53.115328 2023] [proxy_fcgi:error] [pid 179936:tid 548321345920] [client 87.152.111.211:58675] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Mon May 22 20:01:57.365101 2023] [proxy_fcgi:error] [pid 179937:tid 546668802432] (70007)The timeout specified has expired: [client 87.152.111.211:55177] AH01075: Error dispatching request to : (reading input brigade)
[Tue May 23 01:09:29.561702 2023] [proxy_fcgi:error] [pid 299904:tid 547943870848] [client 87.152.111.211:59026] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Tue May 23 01:09:30.901620 2023] [proxy_fcgi:error] [pid 299904:tid 548304560512] [client 87.152.111.211:59026] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Tue May 23 02:10:02.636880 2023] [proxy_fcgi:error] [pid 299905:tid 547432145280] [client 87.152.111.211:59099] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'
[Tue May 23 02:10:05.309585 2023] [proxy_fcgi:error] [pid 299905:tid 547465716096] [client 87.152.111.211:59099] AH01071: Got error 'PHP message: PHP Warning:  Unknown: Input variables exceeded 1000. To increase the limit change max_input_vars in php.ini. in Unknown on line 0'

```

#### Nextcloud Log (inserted last few entries in this gist):
https://gist.github.com/numerfolt/44427da1ba56561d28df9ca2f35ddc58

**Comment by paulijar** [comment]
Extracted from the gist linked to the OP, this is the interesting part of the error message:
```
{"Exception":"Doctrine\\DBAL\\Exception\\DriverException","Message":"An exception occurred while executing a query: SQLSTATE[22007]: Invalid datetime format: 1292 Incorrect datetime value: '55359-03-30 22:30:00.000' for column `nextcloud`.`oc_music_tracks`.`last_played` at row 1","Code":1292,"Trace":[
{"file":"/var/www/nextcloud/3rdparty/doctrine/dbal/src/Connection.php","line":1814,"function":"convert","class":"Doctrine\\DBAL\\Driver\\API\\MySQL\\ExceptionConverter","type":"->"},
{"file":"/var/www/nextcloud/3rdparty/doctrine/dbal/src/Connection.php","line":1749,"function":"handleDriverException","class":"Doctrine\\DBAL\\Connection","type":"->"},
{"file":"/var/www/nextcloud/3rdparty/doctrine/dbal/src/Statement.php","line":184,"function":"convertExceptionDuringQuery","class":"Doctrine\\DBAL\\Connection","type":"->"},
{"file":"/var/www/nextcloud/lib/private/DB/PreparedStatement.php","line":86,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->"},
{"file":"/var/www/nextcloud/apps/music/lib/AppFramework/Db/OldNextcloudMapper.php","line":255,"function":"execute","class":"OC\\DB\\PreparedStatement","type":"->"},
{"file":"/var/www/nextcloud/apps/music/lib/Db/TrackMapper.php","line":383,"function":"execute","class":"OCA\\Music\\AppFramework\\Db\\OldNextcloudMapper","type":"->"},
{"file":"/var/www/nextcloud/apps/music/lib/BusinessLayer/TrackBusinessLayer.php","line":304,"function":"recordTrackPlayed","class":"OCA\\Music\\Db\\TrackMapper","type":"->"},
{"file":"/var/www/nextcloud/apps/music/lib/Controller/SubsonicController.php","line":763,"function":"recordTrackPlayed","class":"OCA\\Music\\BusinessLayer\\TrackBusinessLayer","type":"->"},
{"file":"/var/www/nextcloud/apps/music/lib/Controller/SubsonicController.php","line":177,"function":"scrobble","class":"OCA\\Music\\Controller\\SubsonicController","type":"->"},
```
So it seems that the application tries to save the latest play time to be on year 55359 and the database doesn't like this. Now, it would be interesting to see, what kind of parameters the playSub app gives when it calls the API endpoint [rest/scrobble](http://www.subsonic.org/pages/api.jsp#scrobble). Apparently, it sends the parameters as POST data as the are no urlargs visible on the log lines.

**Comment by numerfolt** [comment]
Do you know of any way to view POST data? I did a quick internet search and couldnt find anything useful. If not, I will just close this as it has nothing to do with the music app but with the app play:Sub.

**Comment by paulijar** [comment]
The only way I can figure out is to make a small modification to the code of the Music app. Not saying that there isn't an easier way, just that I don't know one.

You could add a code block like this:
```
if ($this->request->getMethod() == 'POST') {
	$this->logger->log('POST data: ' . \file_get_contents('php://input'), 'warn');
}
```
...before this line:
https://github.com/owncloud/music/blob/95cec948b6ac7464abb3d2533385afc7617eae9b/lib/Controller/SubsonicController.php#L754

With this, the application should log the POST data on the warning level just before the exception happens.

> ...as it has nothing to do with the music app but with the app play:Sub

Well, it's still a bit early to make this conclusion although I might suspect it.

**Comment by numerfolt** [comment]
Did that, here is the POST data for two of these errors:
```
POST data: p=enc%3A663273396377376177637461&u=USERNAME&c=playSub_iPhoneSE&v=1.9.0&f=json&submission=0&id=track-61924&time=1685048931000
POST data: p=enc%3A663273396377376177637461&u=USERNAME&c=playSub_iPhoneSE&v=1.9.0&f=json&submission=0&id=track-61925&time=1685049190000
```

**Comment by paulijar** [comment]
Okay thanks. I think I have a bug there. The timestamp is given as milliseconds (which matches the API specification) but the Music app interprets it as seconds. This timestamp part of the API call is optional and probably none of the clients I have tested have sent it. That's why this has not been caught earlier.

**Comment by paulijar** [comment]
@numerfolt  Note that in the Subsonic API call arguments, the part following `p=` is your generated API password in a trivially coded format. I censored this part from your post above but you might still want to revoke this and generate a new one.

**Comment by numerfolt** [comment]
Oops, thanks. I changed it 👍

**Comment by paulijar** [comment]
Fix for this has now been released in Music v1.8.4.
